src/adapters/simple_interest_loan_adapter.ts

```diff
@@ -330,6 +334,37 @@ export class SimpleInterestLoanAdapter {
         };
     }
 
+    /**
+     * Asynchronously translates a Dharma debt registry entry into a
+     * simple interest loan order.
+     *
+     * @param entry a Dharma debt registry entry
+     * @return      the translated simple interest loan order
+     */
+    public async fromDebtRegistryEntry(entry: DebtRegistryEntry): Promise<SimpleInterestLoanOrder> {
+        await this.assertIsSimpleInterestTermsContract(entry.termsContract);
+
+        const {
+            principalTokenIndex,
+            principalAmount,
+            interestRate,
+            termLength,
+            amortizationUnit,
+        } = this.termsContractInterface.unpackParameters(entry.termsContractParameters);
+
+        const principalTokenSymbol = await this.contracts.getTokenSymbolByIndexAsync(
+            principalTokenIndex,
+        );
+
+        return {
```

Comment:
Nit, but would be helpful to the reader to construct the specific object that is being returned here: SimpleInterestLoanOrder. Explicit over implicit.